tests/attribute_tests.cpp

```diff
@@ -401,4 +401,48 @@ BOOST_AUTO_TEST_CASE(BytesClone)
     BOOST_CHECK_EQUAL(cs->type(), 19);
 }
 
+BOOST_AUTO_TEST_CASE(ChapPasswordDataConstructor)
+{
+    std::vector<uint8_t> d {'1', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f', 'g'};
+    RadProto::ChapPassword s(3, d.data(), d.size());
+
+    BOOST_CHECK_EQUAL(s.toString(), "31 31323334353637383961626364656667");
+
+    std::vector<uint8_t> values = s.toVector({}, {});
+    std::vector<uint8_t> expected({0x03, 0x13, 0x31, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67});
```

Comment:
This vector almost identical to the `d` vector. Consider making a copy and prepending it with type and length, like you do it for other attribute test.